src/integer_mod_q/mat_polynomial_ring_zq/get.rs

```diff
@@ -156,6 +158,41 @@ impl GetEntry<PolynomialRingZq> for MatPolynomialRingZq {
     }
 }
 
+impl MatPolynomialRingZq {
+    /// Efficiently collects all [`fmpz_poly_struct`]s in a [`MatPolynomialRingZq`] without cloning them.
+    ///
+    /// Hence, the values on the returned [`Vec`] are intended for short-term use
+    /// as the access to [`fmpz_poly_struct`] values could lead to memory leaks or modified values
+    /// once the [`MatPolynomialRingZq`] instance was modified or dropped.
+    ///
+    /// # Examples
+    /// ```compile_fail
+    /// use qfall_math::integer_mod_q::{MatPolynomialRingZq, ModulusPolynomialRingZq};
+    /// use qfall_math::integer::MatPolyOverZ;
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+    /// let poly_mat = MatPolyOverZ::from_str("[[4  -1 0 1 1, 1  42],[2  1 2, 3  1 1 1]]").unwrap();
+    /// let poly_ring_mat = MatPolynomialRingZq::from((&poly_mat, &modulus));
+    ///
+    /// let fmpz_entries = poly_ring_mat.collect_entries();
```

Comment:
Please add a test in the test module, which executes this doc-test to ensure that it's working properly.
Just enter it under the name `doc_test` s.t. anyone every changing the doc-test can/ should be aware to update this one, too.